core/state/database.go

```diff
@@ -109,16 +113,29 @@ func NewDatabase(db ethdb.Database) Database {
 func NewDatabaseWithCache(db ethdb.Database, cache int) Database {
 	csc, _ := lru.New(codeSizeCacheSize)
 	return &cachingDB{
-		db:            trie.NewDatabaseWithCache(db, cache),
+		db: trie.NewDatabaseWithCache(db, cache),
+		// Quorum - Privacy Enhancements
+		privacyMetadataLinker: rawdb.NewPrivacyMetadataLinker(db),
+		// End Quorum - Privacy Enhancements
 		codeSizeCache: csc,
 	}
 }
 
 type cachingDB struct {
-	db            *trie.Database
+	db *trie.Database
+	// Quorum - Privacy Enhancements
+	privacyMetadataLinker rawdb.PrivacyMetadataLinker
+	// End Quorum - Privacy Enhancements
 	codeSizeCache *lru.Cache
```

Comment:
Wrapping with `Quorum - xxx` and `End Quorum - xxx` will result misleading documentation. In particular, `codeSizeCache` will have documentation as `End Quorum - Privacy Enhancements`.

It would be clearer to document the field and its intention in a single line. E.g.:
```go
type cachingDB struct {
	db *trie.Database
	// Quorum:  Privacy enhacements introducing privacyMetadataLinker which maintains mapping between private state root and privacy metadata root. As this struct is the backing store for state, this gives the reference to the linker when needed  
	privacyMetadataLinker rawdb.PrivacyMetadataLinker
```

It can be applied for new methods or  updating existing methods. Giving some context and motivation for the changes.

Reply:
updated the comment you mentioned and a number of other similar comments.